openmc/data/multipole.py

```diff
@@ -129,6 +143,604 @@ def _broaden_wmp_polynomials(E, dopp, n):
+                    found_ideal = True
+                    break
+            else:
+                if log >= DETAILED_LOGGING:
+                    print("  Discarded!")
+
+        if found_ideal:
+            break
+
+        # acceleration
+        if found_better:
+            n_discarded = 0
+        else:
+            if order > max(2*n_peaks, 50) and best_ratio > 0.7:
+                n_discarded += 1
+                if n_discarded >= 10 or (n_discarded >= 5 and best_ratio > 0.9):
+                    if log >= DETAILED_LOGGING:
+                        print("Couldn't get better results. Stop!")
+                    break
+
+    # merge conjugate poles
+    real_idx = []
+    conj_idx = []
+    found_conj = False
+    for i, p in enumerate(best_poles):
+        if found_conj:
+            found_conj = False
+            continue
+        if np.imag(p) == 0.:
+            real_idx.append(i)
+        else:
+            if i < best_poles.size and np.conj(p) == best_poles[i + 1]:
+                found_conj = True
+                conj_idx.append(i)
+            else:
+                raise RuntimeError("Complex poles are not conjugate!")
+    if log:
+        print("Found {} real poles and {} conjugate complex pairs.".format(
+               len(real_idx), len(conj_idx)))
+    mp_poles = best_poles[real_idx + conj_idx]
+    mp_residues = np.concatenate((best_residues[:, real_idx],
+                                  best_residues[:, conj_idx]*2), axis=1)/1j
+    if log:
+        print("Final number of poles: {}".format(mp_poles.size))
+
+    if path_out:
+        if not os.path.exists(path_out):
+            os.makedirs(path_out)
+        for i, mt in enumerate(mts):
+            fig, ax1 = plt.subplots()
+            lns1 = ax1.loglog(test_energy, test_xs_ref[i], 'g', label="ACE xs")
+            lns2 = ax1.loglog(test_energy, best_test_xs[i], 'b', label="VF xs")
+            ax2 = ax1.twinx()
+            lns3 = ax2.loglog(test_energy, best_relerr[i], 'r',
+                              label="Relative error", alpha=0.5)
+            lns = lns1 + lns2 + lns3
+            labels = [l.get_label() for l in lns]
+            ax1.legend(lns, labels, loc='best')
+            ax1.set_xlabel('energy (eV)')
+            ax1.set_ylabel('cross section (b)', color='b')
+            ax1.tick_params('y', colors='b')
+            ax2.set_ylabel('relative error', color='r')
+            ax2.tick_params('y', colors='r')
+
+            plt.title("MT {} vector fitted with {} poles".format(mt, mp_poles.size))
+            fig.tight_layout()
+            fig_file = os.path.join(path_out, "{:.0f}-{:.0f}_MT{}.png".format(
+                                    energy[0], energy[-1], mt))
+            plt.savefig(fig_file)
+            plt.close()
+            if log:
+                print("Saved figure: {}".format(fig_file))
+
+    return (mp_poles, mp_residues)
+
+
+def _vectfit_nuclide(endf_file, njoy_error=5e-4, vf_error=1e-3, vf_pieces=None,
```

Comment:
I think perhaps we should make this function public (no leading underscore). I can imagine a processing workflow where you run this function on all nuclides and then afterward do the windowing for all.

Reply:
Yeah, I did separate the fitting and windowing step in the workflow. Updated.